file/reader_test.go

```diff
@@ -71,6 +72,46 @@ func TestReadKongStateFromStdinFailsToParseText(t *testing.T) {
 	assert.Nil(c)
 }
 
+func TestTransformNotFalse(t *testing.T) {
+	filenames := []string{"-"}
+	assert := assert.New(t)
+	assert.Equal("-", filenames[0])
+
+	var content bytes.Buffer
+	content.Write([]byte("_transform: false\nservices:\n- host: test.com\n  name: test service\n"))
+
+	tmpfile, err := ioutil.TempFile("", "example")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(tmpfile.Name())
+
+	if _, err := tmpfile.Write(content.Bytes()); err != nil {
+		panic(err)
+	}
+
+	if _, err := tmpfile.Seek(0, 0); err != nil {
+		panic(err)
+	}
+
+	oldStdin := os.Stdin
+	defer func() { os.Stdin = oldStdin }() // Restore original Stdin
+
+	os.Stdin = tmpfile
+
+	c, err := GetContentFromFiles(filenames)
```

Comment:
I'd argue that now is the moment when `GetContentFromFiles` deserves refactoring so that the concern of file/stdin reading gets separated from logic.

Reply:
I don't have huge issues with it--it makes the tests a bit odd, but this one was mostly just copy/paste from existing tests, hence some of the unnecessary stuff.

I don't think it should be in scope here: this is just a hotfix to make deck not freak out on something that can appear in Kong configs but essentially changes nothing about its behavior. If we want to change it, it should go into a new issue.

Reply:
Here comes the usual "this is just a hotfix" vs "this is technical debt that makes hotfixes unnecessarily complex" debate :slightly_smiling_face: 

> If we want to change it, it should go into a new issue.

This way technical debt never gets resolved because we keep copy/pasting suboptimal patterns between tests/code sites and punt improvement of these patterns into unspecified future.

That said, understanding our limited capacity at this time, it's not easy to draw the line between code health fixes that are worth fixing when touched and those that aren't.

Let's consider this comment fully optional, but also discuss this issue in a retro; I think it's a quality problem that's gradually driving us into maintenance trouble.